js/reference.js

```diff
@@ -877,18 +906,50 @@
             if (this._canUpdate === undefined) {
                 var ref = (this._context === module._contexts.EDIT) ? this : this.contextualize.entryEdit;
 
-                this._canUpdate = ref._table.kind !== module._tableKinds.VIEW && !ref._table._isGenerated && !ref._table._isImmutable && ref._checkPermissions("update");
+                if (ref._table.kind === module._tableKinds.VIEW) {
+                    this._canUpdate = false;
+                    this._canUpdateReason = "Table is a view. ";
+                } else if (ref._table._isGenerated) {
+                    this._canUpdate = false;
+                    this._canUpdateReason = "Table is generated. ";
+                } else if (ref._table._isImmutable) {
+                    this._canUpdate = false;
+                    this._canUpdateReason = "Table is immutable. ";
+                } else if (!ref._checkPermissions("update")) {
+                    this._canUpdate = false;
+                    this._canUpdateReason = "No permissions to update. ";
+                } else {
+                    this._canUpdate = true;
+                }
 
                 if (this._canUpdate) {
                     var allColumnsDisabled = ref.columns.every(function (col) {
                         return (col.getInputDisabled(module._contexts.EDIT) !== false);
                     });
-                    this._canUpdate = !allColumnsDisabled;
+
+                    if (allColumnsDisabled) {
+                        this._canUpdate = false;
+                        this._canUpdateReason = "All columns are disabled. ";
+                    }
                 }
             }
             return this._canUpdate;
         },
 
+        /**
+         * Indicates the reason as to why a user cannot update for
+         * the referenced resource(s). In some cases, this won't be set
+         * because the user can update, so the value will be `undefined`.
+         * @type {(String|undefined)}
+         */
+        get canUpdateReason() {
+            if (this._canUpdateReason === undefined) {
+                // will set the _canUpdateReason property
+                this.canUpdate();
```

Comment:
canUpdate is not a function, so you cannot call it that way. You have to say `var res = this.canUpdate;`